pandaserver/dataservice/Finisher.py

```diff
@@ -173,23 +174,31 @@ def run(self):
                                         fileNode.appendChild(fsizeNode)
                                         fileNode.appendChild(chksumNode)
                                         topNode.appendChild(fileNode)
-                                # status in file name
+                                # status of the job record
                                 if failedFiles == []:
-                                    statusFileName = 'finished'
+                                    record_status = 'finished'
                                 else:
-                                    statusFileName = 'failed'
+                                    record_status = 'failed'
                                 # write to file
-                                xmlFile = '%s/%s_%s_%s' % (panda_config.logdir,job.PandaID,statusFileName,
-                                                           str(uuid.uuid4()))
-                                oXML = open(xmlFile,"w")
-                                oXML.write(topNode.toxml())
-                                oXML.close()
+                                # xmlFile = '%s/%s_%s_%s' % (panda_config.logdir,job.PandaID,record_status,
+                                #                            str(uuid.uuid4()))
+                                # oXML = open(xmlFile,"w")
+                                # oXML.write(topNode.toxml())
+                                # oXML.close()
+                                # write to job output report table, try update first
+                                tmp_ret = self.taskBuffer.updateJobOutputReport(
+                                    panda_id=job.PandaID, attempt_nr=0, data=topNode.toxml())
+                                if not tmp_ret:
+                                    # then try insert
+                                    self.taskBuffer.insertJobOutputReport(
+                                        panda_id=job.PandaID, prod_source_label=job.prodSourceLabel,
+                                        job_status=record_status, attempt_nr=0, data=topNode.toxml())
```

Comment:
Probably this should be attempt_nr=job.attemptNr. Otherwise, it would be skipped at https://github.com/PanDAWMS/panda-server/blob/bff29ba7dd60bd2fb92690835d83354568752ce1/pandaserver/dataservice/AdderGen.py#L150

To test this kind of thing, usually I use https://github.com/PanDAWMS/panda-server/blob/master/pandaserver/test/finishJob.py
to generate the output report for a job, run add.py so that the job goes to transferring from holding, 
use https://github.com/PanDAWMS/panda-server/blob/master/pandaserver/test/callbackDDM.py to send fake callbacks so that another output report is generated, and then run add.py so that the job goes to finished.